### DISCUSSION TITLE
Show Spotify Track and Artist labels on sketchybar

### DISCUSSION THREAD

**Discussion by bassamsdata** [FelixKratz/SketchyBar#295] [Q&A] [answered]
Hey, I was trying to get the name of the `Track` and `Artist` on the sketchybar and in the popup window using your last Spotify plugin (latest dotfiles) but I couldn't. I messed up things, so sometimes Spotify will not show up at all 😕.
I know there are other spotify plugins that do what I want, but couldn't figure it out to work with your plugin (since I'm not an expert in shell scripting).

Is there anything I would change in order to accomplish that? It was the plugin I was playing with, not the item itself. am I doing it wrong? 🫣

The expected behavior would be similar to the below picture, but with your icon next to the name of the Artist and Track (see below image for an example)

<img width="137" alt="Screen Shot 2022-11-30 at 4 49 32 PM" src="https://user-images.githubusercontent.com/105807570/204917049-911094fc-b89b-45f5-9ab5-67de2fe67544.png">

_image by :_ @neutonfoo

Thanks a lot

**Comment by FelixKratz** [comment, marked as answer]
Hey, you can include the artist and song next to the icon by enabling the `label` of the item displayed in the bar (in this case the `spotify.anchor` in `items/spotify.sh`):
```bash
sketchybar --set spotify.anchor label.drawing=on
```
and then you need to set the label when the track changes (this is done in `plugins/spotify.sh` in the `update` function), e.g.:
```bash
sketchybar --set spotify.anchor label="$ARTIST - $TRACK"
```

**Reply by bassamsdata** [reply]
Thanks a lot, appreciated!
